src/components/SidePanel/SidePanel.jsx

```diff
@@ -0,0 +1,178 @@
+import ReleaseDate from "components/ReleaseDate/ReleaseDate";
+import Keyword from "components/Keyword/Keyword";
+import arrow from "assets/arrow.svg";
+import { Paragraph } from "components/shared-styled-comp/shared-styled-comp";
+/**
+ * Filter and sort component for movies.
+ * @param {object} props - The properties for the component.
+ * @param {function} props.getMovies - Function to retrieve movies.
+ * @param {function} props.setSelectedGenres - Function to set selected genres.
+ * @param {Array} props.selectedGenres - Array of selected genres.
+ * @param {function} props.setAvailabilities - Function to set availabilities.
+ * @param {Array} props.availabilities - Array of availabilities.
+ * @param {function} props.setRelease - Function to set release.
+ * @param {Array} props.release - Array of release types.
+ * @returns {JSX.Element} The SidePanel component.
+ */
+
+export default function SidePanel({
+  getMovies,
+  setSelectedGenres,
+  selectedGenres,
+  setAvailabilities,
+  availabilities,
+  setRelease,
+  release,
+  setSelectedKeyword,
+  selectedKeyword,
+}) {
+  const [isActiveFilter, setIsActiveFilter] = useState(false);
+  const [lang, setLang] = useState([]);
+  const [country, setCountry] = useState([]);
+  const [genres, setGenres] = useState([]);
+
+  /**
+   * Fetches movie genres from the API and sets them using the provided callback.
+   */
+  const getGenres = () => {
+    const apiKey = "4db3b4ee5893cead9657d41699ec4c26";
+    const apiUrl = `https://api.themoviedb.org/3/genre/movie/list?language=en&api_key=${apiKey}`;
+    axios
+      .get(apiUrl)
+      .then((response) => {
+        setGenres(response.data.genres);
+      })
+      .catch((error) => {
+        console.log(error);
+      });
+  };
+
+  /**
+   * Fetches available languages from the API and sets them using the provided callback.
+   */
+  const getLanguages = () => {
+    const apiKey = "4db3b4ee5893cead9657d41699ec4c26";
+    const apiUrl = `https://api.themoviedb.org/3/configuration/languages?api_key=${apiKey}`;
+    axios.get(apiUrl).then((response) => {
+      setLang(response.data);
+    });
+  };
+
+  /**
+   * Fetches available countries from the API and sets them using the provided callback.
+   */
+  const getCountry = () => {
+    const apiKey = "4db3b4ee5893cead9657d41699ec4c26";
+    const apiUrl = `https://api.themoviedb.org/3/configuration/countries?language=en-US&api_key=${apiKey}`;
+    axios.get(apiUrl).then((response) => {
+      setCountry(response.data);
+    });
+  };
+
+  /**
+   * Toggles the display of the content filter and fetches genres, languages, and countries.
+   * @param {object} e - Event object.
+   */
+  const showContentFilter = (e) => {
+    e.preventDefault();
+    setIsActiveFilter((prevIsActive) => !prevIsActive);
+    getGenres();
+    getLanguages();
+    getCountry();
+  };
+
+  return (
+    <div>
+      <form onSubmit={getMovies}>
+        <SortInput />
+        <ContentContainer>
```

Comment:
Can we separate the Filters into their own component similar to what you already have with the `SortInput`. 
It is a good practice to have the same level of granularity of components inside their parent component.